File: src/plugin/command/administrator/PlayerBanCommandPlugin.java
package plugin.command.administrator;

import com.rs.game.entity.actor.player.Player;
import com.rs.game.plugin.type.CommandPlugin;
import com.rs.game.world.punishment.PunishmentHandler;
import com.rs.game.world.punishment.PunishmentType;
import com.rs.utility.game.InputEvent;
import com.rs.utility.game.InputEvent.InputEventType;
import plugin.command.CommandManifest;

/**
 * @author Tyluur <itstyluur@gmail.com>
 * @since 9/13/2017
 */
@CommandManifest(description = "Bans a player", types = { String.class })
public class PlayerBanCommandPlugin extends CommandPlugin {
	
	@Override
	public void handle(Player player, String[] args, boolean console, boolean clientCommand) {
		String name = getCompleted(args, 1);
		player.getPackets().requestClientInput(new InputEvent("Enter Duration (HRS (0 = inf)):", InputEventType.INTEGER) {
			@Override
			public void handleInput() {
				PunishmentHandler.addPunishment(player, name, getInput(), PunishmentType.PLAYER_BAN);
			}
		});
	}
	
	@Override
	public String[] identifiers() {
		return arguments("ban");
	}
}

File: src/main/java/plugin/command/administrator/PlayerBanCommandPlugin.kt
package plugin.command.administrator

import org.redrune.game.content.plugin.type.CommandPlugin
import org.redrune.game.entity.actor.player.Player
import org.redrune.game.global.punishment.PunishmentHandler
import org.redrune.game.global.punishment.PunishmentType
import org.redrune.utility.game.InputEvent
import plugin.command.CommandManifest

/**
 * @author Tyluur <itstyluur@icloud.com>
 * @since 9/13/2017
 */
@CommandManifest(description = "Bans a player", types = [String::class])
class PlayerBanCommandPlugin : CommandPlugin() {
    override fun handle(player: Player, args: Array<String>, console: Boolean, clientCommand: Boolean) {
        val name = getCompleted(args, 1)
        player.packets.requestClientInput(object :
            InputEvent("Enter Duration (HRS (0 = inf)):", InputEventType.INTEGER) {
            override fun handleInput() {
                PunishmentHandler.addPunishment(player, name, getInput(), PunishmentType.PLAYER_BAN)
            }
        })
    }

    override fun identifiers(): Array<String> {
        return arguments("ban")
    }
}